packages/framework-core/src/booster-graphql-dispatcher.ts

```diff
@@ -28,36 +29,37 @@ export class BoosterGraphQLDispatcher {
 
     this.graphQLSchema = new GraphQLGenerator(config, commandDispatcher, this.readModelDispatcher).generateSchema()
     this.websocketHandler = new GraphQLWebsocketHandler(
+      config,
       logger,
-      this.config.provider.readModels.notifySubscription.bind(null, config),
+      this.config.provider.auth,
+      this.config.provider.connections,
       {
         onStartOperation: this.runGraphQLOperation.bind(this),
         onStopOperation: this.readModelDispatcher.unsubscribe.bind(this.readModelDispatcher),
-        onTerminateOperations: this.readModelDispatcher.unsubscribeAll.bind(this.readModelDispatcher),
+        onTerminate: this.handleDisconnect.bind(this),
       }
     )
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public async dispatch(request: any): Promise<any> {
-    const envelope = await this.config.provider.graphQL.rawToEnvelope(request, this.logger)
-    this.logger.debug('Received the following GraphQL envelope: ', envelope)
+  public async dispatch(request: unknown): Promise<unknown> {
+    const envelopeOrError = await this.config.provider.graphQL.rawToEnvelope(request, this.logger)
+    this.logger.debug('Received the following GraphQL envelope: ', envelopeOrError)
 
-    switch (envelope.eventType) {
+    switch (envelopeOrError.eventType) {
       case 'CONNECT':
         return this.config.provider.graphQL.handleResult(null, graphQLWebsocketSubprotocolHeaders)
       case 'MESSAGE':
-        return this.config.provider.graphQL.handleResult(await this.handleMessage(envelope))
+        return this.config.provider.graphQL.handleResult(await this.handleMessage(envelopeOrError))
       case 'DISCONNECT':
-        return this.config.provider.graphQL.handleResult(await this.handleDisconnect(envelope))
+        return this.config.provider.graphQL.handleResult(await this.handleDisconnect(envelopeOrError.connectionID))
       default:
         return this.config.provider.graphQL.handleResult({
-          errors: [new Error(`Unknown message type ${envelope.eventType}`)],
+          errors: [new Error(`Unknown message type ${envelopeOrError.eventType}`)],
         })
     }
   }
 
-  private async handleMessage(envelope: GraphQLRequestEnvelope): Promise<DispatchResult> {
+  private async handleMessage(envelope: GraphQLRequestEnvelope | GraphQLRequestEnvelopeError): Promise<DispatchResult> {
```

Comment:
I'd have used a promise.

Reply:
@itrion Yes, Either is what fits better this place, specially because you can pattern match on it by using [fold](https://gcanti.github.io/fp-ts/modules/Either.ts#fold), but sadly, the team is not comfortable yet using `fp-ts`

@javiertoledo in my opinion, the Promise solution is not very good as it is encoding incorrectly the domain into the type level. A `Promise` denotes execution, and here we're only passing a pure value. Also by using a promise, we are undercovering the intent of the function, which is handing a regular envelope or an envelope containing an error.

Reply:
Explanation of why I didn't use a promise here: https://github.com/boostercloud/booster/pull/373#discussion_r471482391

I tried a lot of things (even creating a specific error) but all of them were messier than this ;-)